src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -224,18 +224,19 @@ public static TypeSymbolWithAnnotations Create(TypeSymbol typeSymbol, bool? isNu
                 return null;
             }
 
-            if (isNullableIfReferenceType == null && typeSymbol.TypeKind == TypeKind.TypeParameter)
+            if (isNullableIfReferenceType != true && typeSymbol.TypeKind == TypeKind.TypeParameter)
             {
-                return new NonLazyType(typeSymbol, isNullable: null, customModifiers);
+                Debug.Assert(customModifiers.IsEmpty);
+                return new LazyNonNullableTypeParameter((TypeParameterSymbol)typeSymbol, isNullableIfReferenceType);
```

Comment:
Nit: Consider renaming `LazyUnannotatedTypeParameter`. "Unannotated" covers both oblivious and `!`.

Reply:
I like it, thanks! I was struggling to come up with a good name.

---
In reply to: [201544285](https://github.com/dotnet/roslyn/pull/28453#discussion_r201544285) [](ancestors = 201544285)